ducktape/services/service.py

```diff
@@ -26,13 +26,14 @@
 # should be large enough to use one instance per service instance.
 
 from ducktape.command_line.config import ConsoleConfig
+from ducktape.template import TemplateRenderer
 
-
-class Service(object):
-    def __init__(self, service_context):
+class Service(TemplateRenderer):
+    def __init__(self, service_context, *args, **kwargs):
         """
         :type service_context: ServiceContext
         """
+        super(Service, self).__init__(*args, **kwargs)
```

Comment:
super() invokes the next constructor in the MRO. This can be affected by the exact set of classes and order the a subclass inherits from this one.  It seems to be a matter of preference whether using super() and multiple inheritence is a good idea. [Here](https://rhettinger.wordpress.com/2011/05/26/super-considered-super/) is one article that gives practical advice about how to use it (this pattern, combined with the keyword-arg stripping approach + the possibility of another mixin/base class in a subclass of Service should explain why I included this here).

This could go either way. There are also people who think it's a [bad idea to use it](https://fuhm.net/super-harmful/). The motivation for using mixins here are two-fold. First, I'm imagining that Service could easily become a behemoth of utility methods, most of which doesn't need to assume much about the class it is mixed into. A lot of those could be kept as separate classes and mixed in as needed by the user, or just kept separate in order to keep things organized internally. I dislike delegation in this particular instance because it would require a ton of boilerplate code just to dispatch calls to the TemplateRenderer we stored as a field. Second, using mixins allows reuse of that code, which I did here by allowing both services and tests to have templates.

Another alternative would be to make TemplateRenderer a standalone class that has to be instantiated to use templates. They just seem like they'll be needed for pretty much every Service, so it seemed worth including via an easy to use method call rather than having to instantiate a different TemplateRenderer manually in every service you write.